bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/api/BondHttpApi.java

```diff
@@ -0,0 +1,266 @@
+     *
+     * @param deviceId The ID of the device
+     * @return the {@link BondDeviceState}
+     * @throws IOException
+     */
+    @Nullable
+    public BondDeviceState getDeviceState(String deviceId) throws BondException {
+        String json = request("/v2/devices/" + deviceId + "/state");
+        logger.trace("BondHome device state : {}", json);
+        try {
+            return gson.fromJson(json, BondDeviceState.class);
+        } catch (JsonParseException e) {
+            logger.debug("Could not parse device {}'s state JSON '{}'", deviceId, json, e);
+            throw new BondException("@text/offline.comm-error.unparseable-response");
+        }
+    }
+
+    /**
+     * Gets the current properties of a device
+     *
+     * @param deviceId The ID of the device
+     * @return the {@link BondDeviceProperties}
+     * @throws IOException
+     */
+    @Nullable
+    public BondDeviceProperties getDeviceProperties(String deviceId) throws BondException {
+        String json = request("/v2/devices/" + deviceId + "/properties");
+        logger.trace("BondHome device properties : {}", json);
+        try {
+            return gson.fromJson(json, BondDeviceProperties.class);
+        } catch (JsonParseException e) {
+            logger.debug("Could not parse device {}'s property JSON '{}'", deviceId, json, e);
+            throw new BondException("@text/offline.comm-error.unparseable-response");
+        }
+    }
+
+    /**
+     * Executes a device action
+     *
+     * @param deviceId The ID of the device
+     * @param actionId The Bond action
+     * @param argument An additional argument for the actions (such as the fan speed)
+     */
+    public synchronized void executeDeviceAction(String deviceId, BondDeviceAction action, @Nullable Integer argument) {
+        String url = "http://" + bridgeHandler.getBridgeIpAddress() + "/v2/devices/" + deviceId + "/actions/"
+                + action.getActionId();
+        String payload = "{}";
+        if (argument != null) {
+            payload = "{\"argument\":" + argument + "}";
+        }
+        InputStream content = new ByteArrayInputStream(payload.getBytes());
+        logger.debug("HTTP PUT to {} with content {}", url, payload);
+
+        final HttpClient httpClient = httpClientFactory.getCommonHttpClient();
+        final Request request = httpClient.newRequest(url).method(HttpMethod.PUT)
+                .header("BOND-Token", bridgeHandler.getBridgeToken())
+                .timeout(BOND_API_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+        try (final InputStreamContentProvider inputStreamContentProvider = new InputStreamContentProvider(content)) {
+            request.content(inputStreamContentProvider, "application/json");
+        }
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (Exception e) {
+            logger.warn("Unable to execute device action {} against device {}: {}", deviceId, action, e.getMessage());
```

Comment:
Would it be worth throwing an exception here to inform caller, which could then log this?